Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1

```diff
@@ -0,0 +1,927 @@
+        Write-Host $([Constants]::SingleDashLine)
+        $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
+        $subId = $controlForRemediation.SubscriptionId
+        $controls = $controlForRemediation.ControlRemediationList
+        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId };
+        $validResources = $resourceDetails.FailedResourceList | Where-Object {![String]::IsNullOrWhiteSpace($_.ResourceId)}
+        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($validResources | Measure-Object).Count -eq 0)
+        {
+            Write-Host "No App Services found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::SingleDashLine)
+            return
+        }
+        $validResources | ForEach-Object { 
+            try
+            {
+                $appServiceResourceAPI = "https://management.azure.com/subscriptions/$subId/resourceGroups/$($_.ResourceGroupName)/providers/Microsoft.Web/sites/$($_.Name)?api-version=2024-04-01" 
+                $AppServiceDetail = Fetch-API -Method Get -Uri $appServiceResourceAPI -ErrorAction Stop                
+                $AppServicesDetails += ($AppServiceDetail | Where-Object { $_.Kind -eq 'app' }) | Select-Object @{N='ResourceId';E={$_.Id}},
+                                                               @{N='ResourceGroupName';E={$_.properties.ResourceGroup}},
+                                                               @{N='Name';E={$_.Name}},
+                                                               @{N='EndToEndEncryptionEnabled';E={$_.Properties.EndToEndEncryptionEnabled}},
+                                                               @{N='Location';E={$_.Location}},
+                                                               @{N='Sku';E={$_.Sku}}
+            }
+            catch
+            {
+                Write-Host "Valid resource information not found in input json file. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host "Skipping the Resource: [$($_.Name)]..."
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName",($_.properties.ResourceGroup))
+                $logResource.Add("Name",($_.Name))
+                $logResource.Add("Reason","Valid ResourceName(s)/ResourceGroupName not found in input json file.")    
+                $logSkippedResources += $logResource
+                Write-Host $([Constants]::SingleDashLine)
+            }
+
+
+        }
+    }
+    else
+    {
+        # No file path provided as input to the script. Fetch all App Services in the Subscription.
+        if ([String]::IsNullOrWhiteSpace($FilePath))
+        {
+            try
+            {
+                Write-Host "Fetching all App Services in Subscription: $($context.Subscription.SubscriptionId)" -ForegroundColor $([Constants]::MessageType.Info)
+
```

Comment:
![image](https://github.com/user-attachments/assets/24fe37d8-0764-4d48-97f7-c1b49aba3d9c)
Can you take reference from the above image of BRS and make changes accordingly?

Reply:
For reference - 
https://github.com/azsk/AzTS-docs/pull/455

Reply:
Done